tools/mason/MasonExternal.chpl

```diff
@@ -324,16 +411,47 @@ proc spkgInstalled(spec: string) {
 
 /* Entry point into the various compiler functions */
 private proc compiler(args: [?d] string) {
-  var option = "list";
-  if args.size > 3 {
-    option = args[3];
-  }
-  select option {
-      when "--list" do listCompilers();
-      when "--find" do findCompilers();
-      when "--edit" do editCompilers();
-      otherwise do masonCompilerHelp();
-    }
+  var parser = new argumentParser();
+  var helpFlag = parser.addFlag(name="help",
+                                opts=["-h","--help"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var listFlag = parser.addFlag(name="list",
+                                opts=["--list"],
+                                defaultValue=true,
+                                flagInversion=false);
+  var findFlag = parser.addFlag(name="find",
+                                opts=["--find"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var editFlag = parser.addFlag(name="edit",
+                                opts=["--edit"],
+                                defaultValue=false,
+                                flagInversion=false);
+  try! {
+    parser.parseArgs(args);
+  }
+  catch ex : ArgumentError {
+    stderr.writeln(ex.message());
+    masonCompilerHelp();
+    exit(1);
+  }
+  if helpFlag.valueAsBool() {
+    masonCompilerHelp();
+    exit(0);
+  }
+  // there was a minor bug in the previous code where the default option of
+  // list was not being selected when no arguments were passed because
+  // the pattern given was list but the pattern to match was --list.
+  // Here we have slightly modified the behavior to match what appeared to be
+  // the intent of the original developers.
+  if findFlag.valueAsBool() {
+    findCompilers();
+  } else if editFlag.valueAsBool() {
+    editCompilers();
+  } else {
```

Comment:
maybe add a comment that this else handles the no-arguments case as well as `--list`